File: libandroid-navigation/src/main/java/org/maplibre/navigation/android/navigation/v5/offroute/OffRoute.java
package org.maplibre.navigation.android.navigation.v5.offroute;

import android.location.Location;

import org.maplibre.navigation.android.navigation.v5.navigation.MapLibreNavigationOptions;
import org.maplibre.navigation.android.navigation.v5.routeprogress.RouteProgress;

public abstract class OffRoute {

  public abstract boolean isUserOffRoute(Location location, RouteProgress routeProgress,
                                         MapLibreNavigationOptions options);
}

File: libandroid-navigation/src/main/java/org/maplibre/navigation/android/navigation/v5/offroute/OffRoute.kt
package org.maplibre.navigation.android.navigation.v5.offroute

import org.maplibre.navigation.android.navigation.v5.location.Location
import org.maplibre.navigation.android.navigation.v5.navigation.MapLibreNavigationOptions
import org.maplibre.navigation.android.navigation.v5.routeprogress.RouteProgress

fun interface OffRoute {

    fun isUserOffRoute(location: Location, routeProgress: RouteProgress, options: MapLibreNavigationOptions): Boolean
}
